core/arch/arm/kernel/secure_partition.c

```diff
@@ -651,29 +678,55 @@ static TEE_Result handle_fdt_mem_regions(struct sp_ctx *ctx, void *fdt)
 		/*
 		 * The SP is a secure endpoint, security attribute can be
 		 * secure or non-secure.
+		 * The SPMC cannot allocate non-secure memory, i.e. if the base
+		 * address is missing this attribute must be secure.
 		 */
-		if (attributes & SP_MANIFEST_ATTR_NSEC)
+		if (attributes & SP_MANIFEST_ATTR_NSEC) {
+			if (alloc_needed) {
+				EMSG("Invalid memory security attribute");
+				return TEE_ERROR_BAD_FORMAT;
+			}
 			is_secure = false;
+		}
+
+		if (alloc_needed) {
+			/* Base address is missing, we have to allocate */
+			mm = tee_mm_alloc(&tee_mm_sec_ddr, size);
```

Comment:
Where is this stored when `handle_fdt_mem_regions()` returns successfully? Or cannot SPs be teared down and have its resources reused?

Reply:
It isn't stored since destroying SPs is not supported yet. Normally the SP is loaded at boot and cannot be closed, but indeed we should add a teardown mechanism in case the SP panics.

Reply:
OK, thanks.